asn1ate/pyasn1gen.py

```diff
@@ -452,7 +458,7 @@ def build_object_identifier_value(self, t):
             elif isinstance(c, NameAndNumberForm):
                 objid_components.append(str(c.number.value))
             else:
-                assert False
+                raise Exception('Unexpected component type %s' % type(c))
```

Comment:
Does this need to be `type(c).__name__` or does str(type(x)) evaluate to the type name?